app/services/freightage_calc.rb

```diff
@@ -0,0 +1,78 @@
+class FreightageCalc
+  attr_reader :distance, :volume
+
+  def initialize(shipping_co:, distance:, weight:, volume:)
+    @shipment_time_settings = shipping_co.shipment_time_settings
+    @price_settings = shipping_co.price_settings
+    @distance = distance
+    @weight = weight
+    @volume = volume
+  end
+
+  def calculate_freightage
+    value = find_price_setting.distance_value * distance
+    days = find_shipment_time_setting.shipping_days
+    { price: value, shipment_days: days }
+  end
+
+  private
+
+  def find_closest
+    @shipment_time_settings.min_by do |setting|
+      difference_min = (setting.distance_min - @distance).abs
+      difference_max = (setting.distance_max - @distance).abs
+
+      difference_max < difference_min ? difference_max : difference_min
+    end
+  end
+
+  def find_price_setting
+    best_price_setting = (price_settings_with_weight & price_settings_with_volume).min_by(&:distance_value)
+
+    return best_price_setting if best_price_setting.present?
+
+    (price_settings_with_weight + price_settings_with_volume).min_by(&:distance_value)
```

Comment:
Esse retorna nulo? se sim daria pra fazer algo como

```suggestion
    best_price_setting || (price_settings_with_weight + price_settings_with_volume).min_by(&:distance_value)
```

Reply:
Se for um array, daria pra usar o presence como embaixo

Reply:
Que interessante... Sim, ele retorna nulo, sua sugestão funciona mto bem. 

Obrigada pela ajuda! :D